java/ql/src/utils/model-generator/CaptureSourceModels.ql

```diff
@@ -0,0 +1,66 @@
+/**
+ * @name Capture source models.
+ * @description Finds APIs that act as sources as they expose already known sources.
+ * @id java/utils/model-generator/sink-models
+ */
+
+import java
+private import Telemetry.ExternalAPI
+private import semmle.code.java.dataflow.DataFlow
+private import semmle.code.java.dataflow.TaintTracking
+private import semmle.code.java.dataflow.ExternalFlow
+private import ModelGeneratorUtils
+private import semmle.code.java.dataflow.internal.FlowSummaryImplSpecific
+private import semmle.code.java.dataflow.internal.FlowSummaryImpl
+private import semmle.code.java.dataflow.internal.DataFlowImplCommon
+private import semmle.code.java.dataflow.internal.DataFlowNodes::Private
+
+class FromSourceConfiguration extends TaintTracking::Configuration {
+  FromSourceConfiguration() { this = "FromSourceConfiguration" }
+
+  override predicate isSource(DataFlow::Node source) { sourceNode(source, _) }
+
+  override predicate isSink(DataFlow::Node sink) {
+    exists(Callable c |
+      sink instanceof ReturnNodeExt and
+      sink.getEnclosingCallable() = c and
+      c.isPublic() and
+      c.fromSource()
+    )
+    or
+    exists(MethodAccess c | sink.asExpr() = c.getAnArgument())
+  }
+
+  override DataFlow::FlowFeature getAFeature() {
+    result instanceof DataFlow::FeatureHasSourceCallContext
```

Comment:
```suggestion
    result instanceof DataFlow::FeatureHasSinkCallContext
```
It's the sink node that's going to end up as our summary, so that's the node that needs to be pluggable in an arbitrary call context.